### ISSUE TITLE
[REQUEST] Overlay configuration for remote modules

### ISSUE THREAD

**Issue by troyready** [onicagroup/runway#263]
What I'd like to be able to do is use a Serverless Framework runway remote module (e.g. this [simple-http-endpoint](https://github.com/serverless/examples/tree/master/aws-node-simple-http-endpoint)) with some additional customization of the module.

E.g., today I can do:

```
deployments:
    - modules:
        - path: git::git://github.com/serverless/examples.git//examples/aws-node-simple-http-endpoint
```

But there's no clear way to add additional customizations to that project. Specifically, something like adding a new CFN resource in that directory's `serverless.yml`.

For CloudFormation modules, `package_sources` (when using the `configs:` option) enables this by way of yaml merging. That level of functionality isn't necessarily needed here, I'd be happy if local files could just be copied over the files in the remote repo.

**Comment by ITProKyle** [comment]
tl;dr from an internal conversation

**runway.yml**

```yaml
deployments:
  - modules:
      - path: git::git://github.com/serverless/examples.git//examples/aws-node-simple-http-endpoint
        options:
          serverless.yml:
            custom:
              new_vars:
                key: value
            resources:
              Resources:
                ExampleNewResource:
                  Type: AWS::Something::Something
                  ...
```

**module deployment workflow**

```
npm ci
npx sls print > <temp-file>.yml
# runway merges the `serverless.yml` option into <temp-file>.yml
npx sls deploy --config <temp-file>.yml
rm -rf <temp-file>.yml
```

if the above is not feasible, a direct merge of the raw yaml will be done.
